axlearn/common/attention.py

```diff
@@ -1748,11 +1757,21 @@ def _forward_for_mode(
                     causal_mask.astype(q_proj.dtype),
                     attention_logit_biases,
                 )
+
+        # Merge segment ids into attention_logit_biases if attention_logit_biases is already set.
+        if attention_logit_biases is not None and segment_ids is not None:
+            attention_logit_biases = apply_attention_logit_biases(
+                make_segment_mask(source_segments=segment_ids, target_segments=segment_ids),
+                attention_logit_biases,
+            )
+            segment_ids = None
```

Comment:
I wonder if this merging should happen in `_compute_attention`. Please see my other comments.